api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -102,7 +102,8 @@ public interface DataFile extends ContentFile<DataFile> {
   int PARTITION_ID = 102;
   String PARTITION_NAME = "partition";
   String PARTITION_DOC = "Partition data tuple, schema based on the partition spec";
-  // NEXT ID TO ASSIGN: 142
+
+  int NEXT_ID_TO_ASSIGN = 142;
```

Comment:
I think I mention this below but I think we don't need this as a constant, since we can just pick the highest value from the base generated for the files table?

Reply:
Changed to find the highest existing id from files table schema, and then assign ids from there